os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -440,6 +440,43 @@ static void _up_assert(int errorcode)
 }
 
 #ifdef CONFIG_BINMGR_RECOVERY
+static void recovery_unblock_task(struct tcb_s *tcb)
+{
+	struct tcb_s *ntcb;
+
+	dq_rem((FAR dq_entry_t *)tcb, (dq_queue_t *)g_tasklisttable[tcb->task_state].list);
+
+	/* Add tcb to ready-to-run list */
+	if (sched_addprioritized(tcb, (FAR dq_queue_t *)&g_readytorun)) {
```

Comment:
You remove the tcb from tasklist and then add it in readytorun. It looks strange.